projects/storefrontlib/src/shared/components/item-counter/item-counter.component.ts

```diff
@@ -70,6 +77,29 @@ export class ItemCounterComponent {
   @HostListener('click') handleClick() {
     this.input.nativeElement.focus();
   }
+  ngOnInit() {
+    if (this.readonly) {
+      this.control.disable({ emitEvent: false });
+    }
+    this.setDisableState();
+
+    this.control$ = this.control.valueChanges.pipe(
+      startWith(this.control.value),
+      tap(value => {
+        // why are we doing this?
+        this.control.setValue(this.getValidCount(value), { emitEvent: false });
+        this.setDisableState();
+      }),
+      map(() => this.control)
+    );
+  }
+
+  protected setDisableState() {
```

Comment:
```suggestion
  protected setDisabledState() {
```
(to align with `disabledState` property)